cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/GetContainerIdOfSegmentCommand.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.segmentstore;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.client.connection.impl.ConnectionPool;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.protocol.netty.WireCommands;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import com.google.common.base.Preconditions;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Gets the container id of the given segment.
+ *  The arguments for the command that is the fully qualified segment name.
+ */
+public class GetContainerIdOfSegmentCommand extends SegmentStoreCommand {
+
+    private final SegmentToContainerMapper segToConMapper;
+
+    public GetContainerIdOfSegmentCommand(CommandArgs args) {
+        super(args);
+        int containerCount = getServiceConfig().getContainerCount();
+        this.segToConMapper = new SegmentToContainerMapper(containerCount, true);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        ConnectionPool pool = createConnectionPool();
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient, pool);
+        final String fullyQualifiedSegmentName = getArg(0);
+        String[] inputParam = fullyQualifiedSegmentName.split("/");
+        Preconditions.checkArgument(inputParam.length == 3, "Invalid qualified-segment-name  '%s'", fullyQualifiedSegmentName);
+        String scope = inputParam[0];
+        String stream = inputParam[1];
+        String[] segmentArr = inputParam[2].split("\\.");
+
+        try {
+            int segmentNumber = Integer.parseInt(segmentArr[0]);
```

Comment:
Not sure I completely understand why we need to get StreamSegmentInfo.
why can't we simply call 
`int containerId = segToConMapper.getContainerId(fullyQualifiedSegmentName);` and be done?

Reply:
This was done to check if the segment exists or not. As previously even some other values were returned for container id for wrong inputs.

Reply:
Can this check be made optional ? for example accepting another parameter called --skipCheck?
It is not always possible that segment store is healthy or that segment even exists. But fro name we should still be able to get the container ID. This check is completely optional and should not prevent use in such cases

Reply:
Right, whether if a segment exists or not may be optional, as we may want to know the container id of arbitrary segments that may not exist (yet).

Reply:
I have added skipCheck